_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+You also should notice the dimming of type parameters to make complicated types easier to read.
+Also the addition of argument names, and showing functions with keyword arguments as they are written, rather than as a weird `#DataFrame#654` internal function.
+The whole thing just looks more modern and polished.
+
+### Time Traveling Debugger
+
+[Julia 1.5 added built-in support](https://julialang.org/blog/2020/05/rr/) for [`rr`](https://rr-project.org/) (Linux only for now).
+This is not useful so much for debugging Julia code, but for debugging Julia itself.
+If you get into one of the very rare situations where the language is doing something truly nonsense; you can send a bug report recorded via `rr`, and someone can debug *exactly* what is happening.
+Event to the extent of [diagnosing faulty RAM](https://julialang.org/blog/2020/09/rr-memory-magic/).
+`rr` is an impressively cool piece of tech.
+It leverages the fact that linux's `fork` is incredibly cheap, because it takes advantage of [copy-on-write](https://en.wikipedia.org/wiki/Copy-on-write).
+`rr` basically `fork`s before every system call, and serializes what is going on; so you have a full track of everything that happened.
+The `rr` integration is really nice and easy to use; I had someone who was part of our architecture and operations team (i.e. who writes way more CloudFormation than Julia) run it to submit a support request; and they had no real troubles.
+
+## Syntax
+
+### `import as`
+
+>The syntax import A as B (plus import A: x as y, import A.x as y, and using A: x as y) can now be used to rename imported modules and identifiers ([#1255](https://github.com/JuliaLang/julia/issues/1255)).
+
+This will be familiar to Python folks who love to write things like: `import numpy as np`.
+I hope we never see it used that ubiquitously in julia.
+I am quite happy with `using Foo` which imports into scope everything the author of `Foo` exported.
+Python people find that super weird and scary; but it's fine.
+Note that in Julia, many (most?) things you use don't belong to the scope of module that defined them anyways; they are overloaded functions e.g. from `Base`.
+So what you import doesn't matter as much as you might think.
+I recently learned that Haskell is like Julia with a default of importing all exports.
+It has been fine for julia, and find for Haskell for ages.
+Though in neither is it required.
+
+The real value of `import FooBar as fb` is not to have a short abbreviation so you can do `fb.quux`.
+That was already possible via `const fb = FooBar`.
+It is to handle cases where the package name itself conflicts with an identifier.
+For example (as has often occurred) if one has uses a `Pipe` in the REPL, and then later wants to load [Pipe.jl](https://github.com/oxinabox/Pipe.jl/) via `import Pipe` then one gets a name clash before it can be even be loaded.
+Now one can do `import Pipe as PipingPipe`.
+
+### NamedTuple/keyword arguments automatic naming
```

Comment:
this could be a good place to mention the type-constructing macro `@NamedTuple{str::String, x::Float64}` if you wanted to include that feature somewhere (ref https://github.com/JuliaLang/julia/pull/34548 and the parser change https://github.com/JuliaLang/julia/pull/34505 to make it possible)

Reply:
Is that in 1.6?

Reply:
it's in 1.5

Reply:
I will add this to the section about standard libary improvements